release-calendar/test/test-data.ts

```diff
@@ -30,59 +30,65 @@ function promoteRelease(
-      new Release('5234567890123'), // nightly
-      new Release('6234567890123'),
-    ].map(async (release) => {
-      return repositoryService.createRelease(release);
-    }),
-  );
+  const releases = [
+    new Release('1234567890123'), // lts
+    new Release('2234567890123'), // stable
+    new Release('3234567890123'), // perc-beta and perc-experimental
+    new Release('4234567890123'), // opt-in-beta and opt-in-experimental
+    new Release('5234567890123'), // nightly
+    new Release('6234567890123'), // rollback
+  ];
 
-  const releases = await repositoryService.getReleases();
-  const promises = [];
-  const startDate = new Date(Date.now());
+  const promotePromises = [];
+  const createPromises = [];
+  const today = new Date(Date.now());
   const channelsForBeta = [
     Channel.NIGHTLY,
     Channel.OPT_IN_BETA,
     Channel.PERCENT_BETA,
     Channel.STABLE,
     Channel.LTS,
   ];
-
-  for (let i = 0; i < channelsForBeta.length - 1; i++) {
-    const newDate = new Date();
-    newDate.setDate(startDate.getDate() + i + 1);
-    const promotions = promoteRelease(
-      releases[0],
-      channelsForBeta[i],
-      channelsForBeta[i + 1],
-      newDate,
-    );
-    promises.push(repositoryService.savePromotions(promotions));
-  }
-
   const channelsForExperimental = [
     Channel.NIGHTLY,
     Channel.OPT_IN_EXPERIMENTAL,
     Channel.PERCENT_EXPERIMENTAL,
   ];
 
-  for (let i = 0; i < channelsForExperimental.length - 1; i++) {
+  for (let i = 0; i < releases.length; i++) {
     const newDate = new Date();
-    newDate.setDate(startDate.getDate() + i + 1);
-    const promotions = promoteRelease(
-      releases[0],
-      channelsForExperimental[i],
-      channelsForExperimental[i + 1],
-      newDate,
-    );
-    promises.push(repositoryService.savePromotions(promotions));
+    newDate.setDate(today.getDate() + i * 5);
+    createPromises.push(repositoryService.createRelease(releases[i], newDate));
+    for (let j = 0; j < channelsForBeta.length - 1 - i; j++) {
+      const promoteDate = new Date();
+      promoteDate.setDate(newDate.getDate() + j + 1);
+      const betaPromotions = promoteRelease(
+        releases[i],
+        channelsForBeta[j],
+        channelsForBeta[j + 1],
+        promoteDate,
+      );
+      promotePromises.push(repositoryService.savePromotions(betaPromotions));
+      if (j < channelsForExperimental.length - 1) {
+        const experimentalPromotions = promoteRelease(
+          releases[i],
+          channelsForExperimental[j],
+          channelsForExperimental[j + 1],
+          promoteDate,
+        );
+        promotePromises.push(repositoryService.savePromotions(experimentalPromotions));
+      }
+    }
   }
 
-  await Promise.all(promises);
+  const rollbackDate = new Date();
+  rollbackDate.setDate(today.getDate() + 5 * 5 + 1);
+  promotePromises.push(repositoryService.savePromotions(promoteRelease(releases[5], Channel.NIGHTLY, Channel.ROLLBACK, rollbackDate)));
```

Comment:
Technically a nightly would never be rolled back (or cherry picked), but a new nightly would be promoted. Feel free to remove this.

Reply:
Hmm... an _old_ nightly might be promoted - e.g. if nightly-wednesday has a P0 in it, we'll promote nightly-monday to take its place. Does that change anything?

Reply:
Database schema supports it 👍